src/app/recipe-detail-page/recipe-detail-page.component.ts

```diff
@@ -1,28 +1,315 @@
+import {MatMenuModule} from '@angular/material/menu';
+import {Recipe} from '../classes/recipe';
+import { Router } from '@angular/router';
+//import { ConfirmDialogComponent } from '../confirm-dialog/confirm-dialog.component';
 
 @Component({
   selector: 'app-recipe-detail-page',
   templateUrl: './recipe-detail-page.component.html',
   styleUrls: ['./recipe-detail-page.component.scss']
 })
-export class RecipeDetailPageComponent /*implements OnInit*/ {
+
+export class RecipeDetailPageComponent implements OnInit {
 
   edit = false;
-  color = "basic";
+  color = "primary";
+  high = 5;
+  highview = 5;
+  highedit = 8;
 
-  constructor() { }
+  recipe_old: Recipe;
+  recipe_new = new Recipe();
 
-  /*ngOnInit(): void {
-  }*/
+  recipe_id: number;
+
+  title: FormControl;
+  author: FormControl;
+  duration: FormControl;
+  nutritional: FormControl;
+  link: FormControl;
+  other: FormControl;
+
+  cookie = "https://raw.githubusercontent.com/MyCookieBook/CookieBookFE/master/src/pictures/Logo.jpg";
+
+  constructor(/*public dialog: MatDialog,*/ private router: Router) {  }
+
+  ngOnInit(): void {
+    this.recipe_id = 0;
+    if(this.recipe_id === 0) {
+      this.recipe_old = new Recipe();
+      this.createEmptyRecipe();
+      this.color = "basic";
+      this.high = this.highedit;
+      this.edit = true;
+    }
+    this.init();
+  }
+
+  init() {
+    this.recipe_new.copy(this.recipe_old);
+
+    this.setFormControl();
+  }
+
+  createEmptyRecipe() {
+    this.recipe_old.setCategoryFE("Other");
+    this.recipe_old.title = "";
+    this.recipe_old.author = "";
+    this.recipe_old.duration = 0;
+    this.recipe_old.nutritional = "";
+    this.recipe_old.difficulty = 0;
+    this.recipe_old.ingredient = [""];
+    this.recipe_old.material = [""];
+    this.recipe_old.step = [""];
+    this.recipe_old.picture = "https://raw.githubusercontent.com/MyCookieBook/CookieBookFE/master/src/pictures/DefaultRecipePicture.jpg";
+    this.recipe_old.link = "";
+    this.recipe_old.other = "";
+  }
+
+  setCategory(category: string) {
+    this.recipe_new.setCategoryFE(category);
+  }
+
+  inputTitle(event) {
+    const title = event.target.value;
+    this.recipe_new.title = title;
+    this.checkInvalid();
+  }
+
+  getErrorMessageTitle() {
+    this.color = "basic"
+    if(this.title.hasError('required')) {
+      return 'Please enter a title for your recipe!';
+    } else if(this.title.hasError('maxlength')) {
+      return 'Your title is too long. (Max length 255)'
+    } else {
+      return '';
+    }
+  }
+
+  inputAuthor(event) {
+    const author = event.target.value;
+    this.recipe_new.author = author;
+    this.checkInvalid();
+  }
+
+  getErrorMessageAuthor() {
+    this.color = "basic"
+    if(this.author.hasError('required')) {
+      return 'Please enter a author for your recipe!';
+    } else if(this.title.hasError('maxlength')) {
+      return 'Your authorname is too long. (Max length 255)'
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCookieBook/CookieBookFE/pullRequest?prid=7556402)
```suggestion
      return "Your authorname is too long. (Max length 255)"
```